src/reconcile/rec_write.c

```diff
@@ -1894,7 +1894,7 @@ __rec_set_page_write_gen(WT_BTREE *btree, WT_PAGE_HEADER *dsk)
      * Other than salvage, the write generation number is used to reset the stale transaction id's
      * present on the page upon server restart.
      */
-    dsk->write_gen = ++btree->write_gen;
+    dsk->write_gen = __wt_atomic_add64(&btree->write_gen, 1);
```

Comment:
I feel like the use of atomic deserves a comment. The code existed without atomic for a long time, and I'd be afraid that someone in the future might think it's not necessary if we don't leave a breadcrumb.

Reply:
Added a comment explaining why we need it to be incremented automatically and also removed an old comment that is no longer true. please check and let me know.

Reply:
lgtm